openwisp_monitoring/utils.py

```diff
@@ -23,7 +23,7 @@ def wrapper(*args, **kwargs):
             try:
                 return method(*args, **kwargs)
             except Exception as err:
-                logger.info(
+                logger.exception(
```

Comment:
This will log the stacktrace, but it will generate alerts on prod systems which we do not want.
Is the goal of this PR to show the stacktrace in the log message or just show the error?

Something like this should work:

```python
err_info = ''.join(traceback.format_exception(type(err), err, err.__traceback__))
logger.info(
  #...
  f'{err_info}'
  #...
)
```
``

Reply:
The objective is to show both. Both exception and stack-trace are important for people that need to resolve an issue in their wrapped code. We can have a higher level default log level (logging.CRITICAL) be the default log level and that would solve alerts being generated? This can be user-configurable through the settings API?

I'm currently talking to a user that _is getting spammed_ with alerts but there is no actionable feedback on which part of the application (monitoring backend, protocol transport client, ...) is actually at fault.

Reply:
ok, a solution similar to the one suggested above will show the stack trace.
In this particular case, we do not want to send alerts for errors that can be retried. We only want to send alerts when there is a problem we can fix. If there's a momentary network error we do not need to be alerted, logging as info is fine.

In the particular case of the user who is getting spammed, are you sure this is the piece of code which is causing the spamming of alerts? This code does not generate any alert unless the retry mechanism gives up.

Reply:
If you see at the end of the function, if the max attempts are reached, the exception is raised, so there's no need to log exceptions here. If the problem is persistent, it will be raised and logged as error.